<?xml version="1.0" ?>
<robot name="Universal Robot Webots">
    <webots>
        <plugin type="webots_ros2_control::Ros2Control" />
    </webots>

    <ros2_control name="WebotsControl" type="system">
        <hardware>
            <plugin>webots_ros2_control::Ros2ControlSystem</plugin>
        </hardware>

        <!-- Universal Robot -->
        <joint name="shoulder_pan_joint">
            <state_interface name="position"/>
            <command_interface name="position"/>
        </joint>
        <joint name="shoulder_lift_joint">
            <state_interface name="position"/>
            <command_interface name="position"/>
        </joint>
        <joint name="elbow_joint">
            <state_interface name="position"/>
            <command_interface name="position"/>
        </joint>
        <joint name="wrist_1_joint">
            <state_interface name="position"/>
            <command_interface name="position"/>
        </joint>
        <joint name="wrist_2_joint">
            <state_interface name="position"/>
            <command_interface name="position"/>
        </joint>
        <joint name="wrist_3_joint">
            <state_interface name="position"/>
            <command_interface name="position"/>
        </joint>

        <!-- ROBOTIQ 3F Gripper -->
        <joint name="palm_finger_1_joint">
            <state_interface name="position"/>
            <command_interface name="position"/>
        </joint>
        <joint name="finger_1_joint_1">
            <state_interface name="position"/>
            <command_interface name="position"/>
        </joint>
        <joint name="finger_1_joint_2">
            <state_interface name="position"/>
            <command_interface name="position"/>
        </joint>
        <joint name="finger_1_joint_3">
            <state_interface name="position"/>
            <command_interface name="position"/>
        </joint>
        <joint name="palm_finger_2_joint">
            <state_interface name="position"/>
            <command_interface name="position"/>
        </joint>
        <joint name="finger_2_joint_1">
            <state_interface name="position"/>
            <command_interface name="position"/>
        </joint>
        <joint name="finger_2_joint_2">
            <state_interface name="position"/>
            <command_interface name="position"/>
        </joint>
        <joint name="finger_2_joint_3">
            <state_interface name="position"/>
            <command_interface name="position"/>
        </joint>
        <joint name="finger_middle_joint_1">
            <state_interface name="position"/>
            <command_interface name="position"/>
        </joint>
        <joint name="finger_middle_joint_2">
            <state_interface name="position"/>
            <command_interface name="position"/>
        </joint>
        <joint name="finger_middle_joint_3">
            <state_interface name="position"/>
            <command_interface name="position"/>
        </joint>

    </ros2_control>
</robot>
